python/lsst/daf/butler/core/dimensions/_coordinate.py

```diff
@@ -957,7 +956,7 @@ def hasRecords(self) -> bool:
 
     def _record(self, name: str) -> DimensionRecord | None:
         # Docstring inherited from DataCoordinate.
-        assert False
```

Comment:
This looks wrong to me and `AssertionError` doesn't seem any better. Is it meant to be `NotImplementedError`?